aiida/orm/querybuilder.py

```diff
@@ -409,168 +124,90 @@ def __init__(
+        self._tags = _QueryTagMap()
+
+        # Set the debug level
         self.set_debug(debug)
 
-        # One can apply the path as a keyword. Allows for jsons to be given to the QueryBuilder.
+        # Validate & add the query path
         if not isinstance(path, (list, tuple)):
             raise TypeError('Path needs to be a tuple or a list')
-        # If the user specified a path, I use the append method to analyze, see QueryBuilder.append
         for path_spec in path:
             if isinstance(path_spec, dict):
                 self.append(**path_spec)
             elif isinstance(path_spec, str):
-                # Maybe it is just a string,
-                # I assume user means the type
+                # Assume user means the entity_type
                 self.append(entity_type=path_spec)
             else:
-                # Or a class, let's try
                 self.append(cls=path_spec)
-
-        # Projections. The user provides a dictionary, but the specific checks is
-        # left to QueryBuilder.add_project.
+        # Validate & add projections
         projection_dict = project or {}
         if not isinstance(projection_dict, dict):
             raise TypeError('You need to provide the projections as dictionary')
         for key, val in projection_dict.items():
             self.add_projection(key, val)
-
-        # For filters, I also expect a dictionary, and the checks are done lower.
+        # Validate & add filters
         filter_dict = filters or {}
         if not isinstance(filter_dict, dict):
             raise TypeError('You need to provide the filters as dictionary')
         for key, val in filter_dict.items():
             self.add_filter(key, val)
-
-        # The limit is caps the number of results returned, and can also be set with QueryBuilder.limit
+        # Validate & add limit
         self.limit(limit)
-
-        # The offset returns results after the offset
+        # Validate & add offset
         self.offset(offset)
-
-        # The user can also specify the order.
-        self._order_by: List[dict] = []
+        # Validate & add order_by
         if order_by:
             self.order_by(order_by)
 
-    def as_dict(self) -> QueryDict:
+    def as_dict(self, copy: bool = True) -> QueryDictType:
         """Convert to a JSON serialisable dictionary representation of the query."""
-        return copy.deepcopy({  # type: ignore[return-value]
+        data: QueryDictType = {
             'path': self._path,
             'filters': self._filters,
             'project': self._projections,
             'order_by': self._order_by,
             'limit': self._limit,
             'offset': self._offset,
-        })
+            'distinct': self._distinct,
+        }
+        if copy:
+            return deepcopy(data)
+        return data
 
     @property
-    def queryhelp(self) -> QueryDict:
+    def queryhelp(self) -> 'QueryDictType':
```

Comment:
Shouldn't we just deprecate this so we can remove it at some point? No reason to keep this around if there is a preferred alternative

Reply:
Yeh I mentioned this here: https://github.com/aiidateam/aiida-core/pull/5081#discussion_r689362848, lets open an issue for it to track

Reply:
#5096

Reply:
Ah yeah, I remember seeing this now. I am a bit surprised by you being against having deprecation warnings for a new major release? Why would that be objectionable? And why wouldn't that be a problem on a minor release?